apitools/base/protorpclite/messages.py

```diff
@@ -250,8 +251,7 @@ class MyMessage(Message):
         outer_definition_name = cls.outer_definition_name()
         if outer_definition_name is None:
             return six.text_type(cls.__name__)
-        else:
-            return u'%s.%s' % (outer_definition_name, cls.__name__)
+        return u'%s.%s' % (outer_definition_name, cls.__name__)
```

Comment:
I approve of the linter. The idea is you want to return early from functions and not have nested if-elses that are hard to follow. This also means converting "if A return; elseif B return; else return;" to "if A return; if B return; return;". Notice the lack of elses. It makes each individual condition stand on its own without linking it to something above. Imagine having to break condition B into B1 and B2 and things get messy fast in the first example.
Some more info on this from other people - https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement